### ISSUE TITLE
Use webpack for node-debug2

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#59237]
This is the continuation of https://github.com/Microsoft/vscode/issues/57680. This is different from builtin extensions, a sample can be found here: https://github.com/Microsoft/vscode-extension-samples/tree/master/webpack-sample

**Comment by roblourens** [comment]
Just a heads up @digeff, I am working on this to package node-debug2 with webpack. It shouldn't change anything about the debug adapter is run. But let me know if you have any concerns.

**Comment by roblourens** [comment]
@dbaeumer The webpack loader for vscode-nls rewrites the localize calls in the bundled js but it doesn't write out the string json bundles, is there any way to make it do that, or should I keep gulp for that?

Also, I'm not sure of the right way to manage the vscode-chrome-debug-core npm module. It is also localized with vscode-nls. If it's bundled along with node2, I should just have to copy its loc bundles out of node_modules/vscode-chrome-debug-core and put them next to the .js bundle. But, they have the same names as node2's json bundles. So, we would need to process the -core loc calls again to add some namespace or another lookup path or something. Or, merge the loc bundles, which we can't really do because there will be file name conflicts.

If I can't bundle -core, there's no point in doing this because that's where 95% of the code is. So the other option is to webpack -core separately, and publish the bundled module, and don't bundle it with node2. I think that would work, but it seems best if a library isn't bundled, and only the end user is bundled along with its dependencies. Not sure what the best practice is.

Any other ideas?

**Comment by dbaeumer** [comment]
I discussed this with @jrieken and he will have a look at this. The end goal should be that the nls.metadata.json files are inlined into the pack. There is no need to keep them separate.

**Comment by jrieken** [comment]
> webpack loader for vscode-nls rewrites the localize calls in the bundled js but it doesn't write out the string json bundles,

I can add a stream to vscode-nls-dev that does this. Today, `rewriteLocalizeCalls` does two things: 1) rewrite the calls to localise and 2) generate those meta-data-files. I will create a separate stream for 2) so that the gulp story stays complete...

> Also, I'm not sure of the right way to manage the vscode-chrome-debug-core npm module. It is also localized with vscode-nls. 

I think the easiest would be to bundle that module before publishing it. That would ensure the nls files are inside the module files...

**Comment by jrieken** [comment]
> webpack loader for vscode-nls rewrites the localize calls in the bundled js but it doesn't write out the string json bundles,

Actually, after taking a second look... this is possible. I can add a flag to the loader to emit those json-files so that the downlevel gulp tasks keep working

**Comment by weinand** [comment]
We've converted the "package" and "publish" npm scripts of node-debug (legacy) to use webpack.

Since I could not create reliable sourcemaps via webpack's ts-loader, I've decided to ship webpacks without sourcemaps and to use the old build scripts for development. A consequence is that source maps are only available for development and nls support only for production.

This [commit](https://github.com/Microsoft/vscode-node-debug/commit/c247d0bcc1a75b3d62d18c854be344cfb97243b5) contains the full webpack transition and this [commit](https://github.com/Microsoft/vscode-node-debug/commit/0cd54245a92f2e2c0a2ecf43fcaabf44dd8cb781) removes the unnecessary i18n-files.

Left to do for November:
- convert node-debug2
- fix the bad source maps created by the ts-loader
